blocks/market-insight/market-insight.js

```diff
@@ -0,0 +1,209 @@
+import { readBlockConfig, fetchPlaceholders } from '../../scripts/aem.js';
+import { createDiv } from '../../scripts/blocks-utils.js';
+
+// TODO: This is dummy function that fetch sample data from EDS json.
+// It will be replaced when API call is available.
+async function fetchMarketInsightMockData() {
+  const hostUrl = window.location.origin;
+  const apiUrl = `${hostUrl}/draft/jiang/marketinsight.json`;
+  try {
+    const response = await fetch(apiUrl);
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
+    const data = await response.json();
+    const results = data.data.map((result) => ({
+      title: result.title,
+      description: result.description,
+      link: result.link,
+      publishedon: result.publishedon,
+    }));
+    return results;
+  } catch (error) {
+    return [];
+  }
+}
+
+function decorateTitle(blockCfg) {
+  const { title } = blockCfg;
+  const blockTitleDiv = createDiv('div', 'title');
+  const blockTitle = createDiv('h2', '');
+  blockTitle.textContent = title;
+  blockTitleDiv.append(blockTitle);
+  return blockTitleDiv;
+}
+
+function decorateDiscoverMore(blockCfg, placeholders) {
+  const discoverMore = blockCfg['discover-more-link'];
+  const discoverMoreDiv = createDiv('div', 'discover-more');
+  const aLink = createDiv('a', '');
+  aLink.target = '_blank';
+  aLink.href = discoverMore;
+  aLink.textContent = placeholders.discovermore;
+  aLink.classList.add('discover-more-button');
+  discoverMoreDiv.append(aLink);
+  return discoverMoreDiv;
+}
+
+function createSocialButton(button, block) {
+  const link = button.closest('li').querySelector('a').href;
+  const encodeLink = encodeURIComponent(link);
+  const encodeTitle = encodeURIComponent(document.title);
+  const socialMap = new Map([
+    ['whatsapp', `https://api.whatsapp.com/send?text=Hey! Check out this: ${encodeLink}`],
+    ['facebook', `http://www.facebook.com/sharer.php?u=${encodeLink}&t=${encodeTitle},'sharer',toolbar=0,status=0,width=626,height=436`],
+    ['linkedin', `https://www.linkedin.com/sharing/share-offsite/?url=${link}`],
+    ['twitter', `https://twitter.com/intent/tweet?url=${link}`],
+  ]);
+
+  const alinks = block.querySelectorAll('.modal .modal-body li a');
+  if (alinks && alinks.length > 0) {
+    [...alinks].forEach((alink) => {
+      if (!alink.classList.contains('copy-link')) {
+        alink.href = socialMap.get(alink.className);
+      }
+    });
+  }
+}
+
+function addSocialButtonEvent(button, block) {
+  button.addEventListener('click', () => {
+    const modal = block.querySelector('.modal');
+    if (modal) {
+      createSocialButton(button, block);
+      modal.classList.toggle('visible');
+      const body = document.querySelector('body');
+      body.classList.toggle('modal-open');
+    }
+  });
+}
+
+function decorateCards(results, blockCfg, block) {
+  const powerBy = (blockCfg['power-by'] ?? '').trim().toLowerCase();
```

Comment:
this can come from placeholder, since it is something repeated at many blocks

Reply:
Agree, I missed it.